crates/ra_hir/src/nameres/raw.rs

```diff
@@ -67,12 +68,19 @@ impl RawItems {
         db: &(impl DefDatabase + AstDatabase),
         file_id: HirFileId,
     ) -> (Arc<RawItems>, Arc<ImportSourceMap>) {
+        let crate_graph = db.crate_graph();
+        let cfg_options = match crate_graph.crate_id_for_crate_root(file_id.original_file(db)) {
+            None => Cow::Owned(CfgOptions::default()),
+            Some(crate_id) => Cow::Borrowed(crate_graph.cfg_options(crate_id)),
+        };
```

Comment:
I think it's better to push cfg handlining slightly higher, to the `DefCollector` layer. That would allow, for example, to reuse the same `RawItems` for both test and non-test versions of the crate. I imagine that, at raw items layer, we collect attributes into an `mbe::TokenTree` and store it with the RawItem. In the `DefCollector`, we parse this token tree and, if that's a `cfg`, we check it and possible skip the item.